### PR TITLE
[CALCITE] Iteration Statement: FOR Statement

### PR DESCRIPTION
**Pull request by ArthurInTheShell**
Added support for For Statement with tests.

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/sql/SqlForStmt.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  public final SqlNode cursorSpecification;
+
+  /**
+   * Creates a {@code SqlForStmt}.
+   * @param pos         Parser position, must not be null.
+   * @param statements  List of statements to iterate, must not be null.
+   * @param beginLabel  Optional begin label, must match end label if not null.
+   * @param endLabel    Optional end label, must match begin label if not null.
+   * @param forLoopVariable
+   *                    The name of the loop.
```
Comment (dasch-1) [inline core/src/main/java/org/apache/calcite/sql/SqlForStmt.java:43]: nit: Should this be "loop variable"?

Comment (dasch-1) [reply core/src/main/java/org/apache/calcite/sql/SqlForStmt.java:43]: Also, should we say that it "must not be null"?

File: core/src/main/java/org/apache/calcite/sql/SqlForStmt.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+   * @param cursorName  The name of the cursor.
+   * @param cursorSpecification
+   *                    A single select statement used as the cursor.
```
Comment (dasch-1) [inline core/src/main/java/org/apache/calcite/sql/SqlForStmt.java:46]: nit: "must not be null"?

File: core/src/main/java/org/apache/calcite/sql/SqlForStmt.java (hunk 1 above)
Comment (ArthurInTheShell) [reply core/src/main/java/org/apache/calcite/sql/SqlForStmt.java]: Updated in https://github.com/googleinterns/calcite/pull/295/commits/0729b1498ee2b87bc6f1697d05000da7b7fb4228

File: core/src/main/java/org/apache/calcite/sql/SqlForStmt.java (hunk 2 above)
Comment (ArthurInTheShell) [reply core/src/main/java/org/apache/calcite/sql/SqlForStmt.java]: Added in https://github.com/googleinterns/calcite/pull/295/commits/0729b1498ee2b87bc6f1697d05000da7b7fb4228

**Pull request merged**